app/api-v2/routes/run-process.js

```diff
@@ -1,124 +1,129 @@
+            if (output.parent_index > request.inputs.length) {
+              logger.trace(`Parent index out of range`)
+              res.status(400).json({ message: `Parent index out of range` })
+              return
+            }
+            if (parentIndices.has(output.parent_index)) {
+              logger.trace(`Duplicate parent index used`)
+              res.status(400).json({ message: `Duplicate parent index used` })
+              return
+            }
+            parentIndices.add(output.parent_index)
+          }
 
-          let result
           try {
-            result = await apiService.runProcess(request.inputs, outputs)
+            return {
+              roles: await processRoles(output.roles),
+              metadata: await processMetadata(output.metadata, req.files),
+              parent_index: output.parent_index,
+            }
           } catch (err) {
-            logger.error(`Unexpected error running process: ${err}`)
-            res.status(500).json({
-              message: `Unexpected error processing items`,
-            })
-          }
-
-          res.status(200).json(result)
-        } catch (err) {
-          logger.error(`Error running process. Error was ${err.message || JSON.stringify(err)}`)
-          if (!res.headersSent) {
-            res.status(500).send(`Error running process`)
+            logger.trace(`Invalid outputs: ${err.message}`)
+            res.status(400).json({ message: err.message })
             return
           }
-        }
-      })
+        })
+      )
+
+      let result
+      try {
+        result = await apiService.runProcess(request.inputs, outputs)
+      } catch (err) {
+        logger.error(`Unexpected error running process: ${err}`)
+        res.status(500).json({
+          message: `Unexpected error processing items`,
+        })
+      }
+
+      res.status(200).json(result)
     },
   }
 
   doc.POST.apiDoc = {
     summary: 'Governs the creation and destruction of all tokens in the system',
-    parameters: [
-      {
-        description: 'Inputs to be created within the system',
-        in: 'body',
-        required: true,
-        name: 'request',
-        allowEmptyValue: true,
-        schema: {
-          properties: {
-            request: {
-              properties: {
-                inputs: {
-                  items: {
-                    $ref: '#/components/schemas/Input',
-                  },
-                },
-                outputs: {
-                  items: {
-                    $ref: '#/components/schemas/Output',
-                  },
+    requestBody: {
+      content: {
+        'multipart/form-data': {
+          schema: {
+            type: 'object',
+            properties: {
+              files: {
+                type: 'array',
+                items: {
+                  type: 'string',
+                  format: 'binary',
                 },
               },
+              request: {
+                type: 'string',
```

Comment:
shouldn't this be an object with properties?

Reply:
It should be but I just can't get it working as an object. Something about OpenAPI, it doesn't handle JSON within multipart/form-data requests properly.
![image](https://user-images.githubusercontent.com/40722025/148221800-271692a9-1983-42a9-ad4c-64e47a7b5243.png)

Reply:
this is even with setting the encoding for `request` as `application/json`

Reply:
will run this locally and take a look...

Reply:
I vote we leave this for now since we'll be splitting up `/run-process` in the future anyway. It works with `request` just as a `string`

Reply:
Sure, happy to approve this for now.